### PR TITLE
[FIX] Finally fixes the stuttering that was caused due to miscalculation in offsets

### PR DESCRIPTION
**Pull request by KoloInDaCrib**
## Does this PR close any issues? If so, link them below.
[3667](https://github.com/FunkinCrew/Funkin/issues/3667) + like 3 others that were closed due to being duplicates

## Briefly describe the issue(s) fixed.
Considering traces are back (hooray!) I was able to determine the culprit of the "minuscule" lags, ty eric
image

so yeah i changed the difference check from 5ms to 10ms, hopefully there shouldn't be any more input errors until another update to offsets lol
Rest in Peace the "Swing Mode" glitch saga you were too funny to last

## Include any relevant screenshots or videos.
Before:
https://github.com/user-attachments/assets/23cee938-390c-40fe-b9b5-f8260c913adc

After:
https://github.com/user-attachments/assets/39023c8f-c87d-49cb-862a-128a9b6500c9

### CODE REVIEW

**Comment by NotHyper-474** [comment]
This is so peak

**Comment by biomseed** [comment]
It's not caused primarily by offsets so that issue should not be closed

**Review by biomseed — APPROVED** [review]

**Comment by trayfellow** [comment]
Can we have more people test this to make sure it's finally fixed fr this time?

**Comment by JustKolosaki** [comment]
> Can we have more people test this, even while also messing with the input offset settings to make sure it's finally fixed fr this time?

You can just compile the game with these changes and test it out yourself - I'll definitely do a bit more testing myself and archive this if theres more errors than I could fix

**Comment by biomseed** [comment]
> I should probably make all the insts and vocals get precached during the preload sequence at some point.

I think that isn't in the scope of this PR

**Comment by JustKolosaki** [comment]
> I think that isn't in the scope of this PR

ye i meaint in another pr lol, but yeah no offset math should now be of any problem as i have tested a bunch of songs both with additional offsets and without em

**Comment by doggogit** [comment]
> so yeah i changed the difference check from 5ms to 10ms, hopefully there shouldn't be any more input errors until another update to offsets lol Rest in Peace the "Swing Mode" glitch saga you were too funny to last

Is this the way to reproduce the bug? If so, I can `gh checkout` the PR and see if it's fixed on my end..
If there's anything else you want me to replicate, please do let me know 👍

**Comment by JustKolosaki** [comment]
> Is this the way to reproduce the bug? If so, I can `gh checkout` the PR and see if it's fixed on my end.. If there's anything else you want me to replicate, please do let me know 👍

Literally just play any song. If you notice any stuttering during the song where the notes would just suddenly move backwards, thats the bug that I was trying to fix

**Comment by doggogit** [comment]
> > Is this the way to reproduce the bug? If so, I can `gh checkout` the PR and see if it's fixed on my end.. If there's anything else you want me to replicate, please do let me know 👍
> 
> Literally just play any song. If you notice any stuttering during the song where the notes would just suddenly move backwards, thats the bug that I was trying to fix

Sorry, let me rephrase, I wanted to ask what you set the input offset to

**Comment by JustKolosaki** [comment]
> Sorry, let me rephrase, I wanted to ask what you set the input offset to, didn't mean to overcomplicate my question

Oh, this bug was ongoing for any offset, including 0

**Comment by doggogit** [comment]
> > Sorry, let me rephrase, I wanted to ask what you set the input offset to, didn't mean to overcomplicate my question
> 
> Oh, this bug was ongoing for any offset, including 0

Oh, huh, weird, I haven't had that bug on 0.5.2

**Comment by biomseed** [comment]
> Oh, huh, weird, I haven't encountered that bug on 0.5.2 or 0.5.1 as a matter of fact

Try a different FPS limit, preferably lower

**Review by biomseed — COMMENTED** [review]
If the stutter varies from song to song, then modded songs could still be affected

**Comment by trayfellow** [comment]
> If the stutter varies from song to song, then modded songs could still be affected

Agreed, if Fresh pico mix caused a difference leading to the PR getting edited again, then there's no telling what other modded songs could do

**Comment by M7theguy** [comment]
this should get the high priority tag

**Comment by biomseed** [comment]
> Agreed, if Fresh pico mix caused a difference leading to the PR getting edited again, then there's no telling what other modded songs could do

This also applies to new songs added in the base game

File: source/funkin/play/PlayState.hx [hunk 1]
```diff
@@ -1477,7 +1477,9 @@ class PlayState extends MusicBeatSubState
       }
 
       if (!startingSong
-        && (Math.abs(FlxG.sound.music.time - correctSync) > 5 || Math.abs(playerVoicesError) > 5 || Math.abs(opponentVoicesError) > 5))
+        && (Math.abs(FlxG.sound.music.time - correctSync) > 15
+          || Math.abs(playerVoicesError) > 15
+          || Math.abs(opponentVoicesError) > 15)) // 5ms was too little according to the command prompt that says it should be 15ms, my bad gang!
```
Comment (AbnormalPoof) [inline source/funkin/play/PlayState.hx:1482]: Tiny nitpick but this comment is not really needed.

Side note: I really hope this is tested by other people so it's fixed for real this time.

**Comment by Beelijah** [comment]
yes dude this is the only reason ive been playing on v0.5.0 instead of the latest (v0.5.3)

**Comment by charlesisfeline** [comment]
NOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO
